pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java

```diff
@@ -318,4 +298,40 @@ private static void generateManagedLedgerBookieClientMetrics(PulsarService pulsa
             // nop
         }
     }
+
+
+    private static void writeOutMetricData(OutputStream out, ByteBuf buffer, boolean exposeBufferMetrics,
+                                           boolean enableCompress) throws IOException {
+        try {
+            if (enableCompress) {
+                out = new GZIPOutputStream(out);
```

Comment:
I think `GZIPOutputStream` will create a lot of garbage on the JVM heap :| 

Could we just use `org.apache.pulsar.common.compression.CompressionCodecZLib` on the buffer?

Reply:
@merlimat  But `CompressionCodecZLib` not in GZIP format, and Prometheus read metrics data in GZIP format:
```java

	if s.gzipr == nil {
		s.buf = bufio.NewReader(resp.Body)
		s.gzipr, err = gzip.NewReader(s.buf)
		if err != nil {
			return "", err
		}
	} else {
		s.buf.Reset(resp.Body)
		if err = s.gzipr.Reset(s.buf); err != nil {
			return "", err
		}
	}

	n, err := io.Copy(w, io.LimitReader(s.gzipr, s.bodySizeLimit))
	s.gzipr.Close()
	if err != nil {
		return "", err
	}
```